### ISSUE TITLE
Error while installing the module

### ISSUE THREAD

**Issue by ottiker** [adioo/flow-auth#3]
What do we have to write to provide the `root user account`? And I think a promt while installing npm modules, will complicate application installing a lot. Maybe there's a different solution.

``` sh
Please provide the root user account [service@jillix.com]: 1234
scripts/ask_for_root_credentials.sh: 16: scripts/ask_for_root_credentials.sh: Syntax error: "(" unexpected (expecting "then")

npm ERR! Linux 3.16.0-34-generic
npm ERR! argv "/usr/local/bin/iojs" "/usr/local/bin/npm" "install"
npm ERR! node v1.6.2
npm ERR! npm  v2.7.1
npm ERR! code ELIFECYCLE
npm ERR! service@0.1.0 install: `scripts/ask_for_root_credentials.sh`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the service@0.1.0 install script 'scripts/ask_for_root_credentials.sh'.
npm ERR! This is most likely a problem with the service package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     scripts/ask_for_root_credentials.sh
npm ERR! You can get their info via:
npm ERR!     npm owner ls service
npm ERR! There is likely additional logging output above.
```

**Comment by gabipetrovay** [comment]
Probably the regex syntax or the `if` with regex was not supported by your default scripting engine. For the above error I see two potential problems that might have caused it:
1. Missing shebang (fixed in [b3eda2f](https://github.com/jillix/service/commit/b3eda2fe9d9f53afa7d7eddaaa13690961348924))
2. A different scripting engine on your machine (point 1 makes sure now it is `bash`) or older `bash` version.

Does this now work for you?

**Comment by gabipetrovay** [comment]
Now the logic behind: it is a little simpler to provide your own credentials or to not provide them at all if they are already present, rather than knowing the internals of the module (where to put them, in what format, etc.). Moreover the installation of other Unix server packages (like postfix or MySQL) can not be completed without the user providing an input (host name, root user, etc.).

What is the alternative to the prompt? I don't see hardcoding it as an alternative (there is already a default for the user account).

On the server, `service` can not be installed through `service`. So as a first application, it must be deployed manually. And in this process, probably not only this root user account but other required data will be necessary.

**Comment by ottiker** [comment]
Now I get this error:

``` sh
Please provide the root user account [service@jillix.com]: service@jillix.com
This should look like an email address!
```

**Comment by ottiker** [comment]
If prompts are inevitable, we need a UI solution in the service. Otherwise the service can not complete an application installation.

**Comment by IonicaBizau** [comment]
I :-1: using bash scripts in installation steps. I'd vote to use OS agnostic NodeJS code which will work on all platforms.

That's another reason why following the git workflow is important.

**Comment by gabipetrovay** [comment]
@IonicaBizau Do we have to use OS agnostic scripts if we only intend to deploy the service on UNIX-based infrastructure?
@adioo but the service has to be deployed first before one can use its UI.

**Comment by IonicaBizau** [comment]
@gabipetrovay I would say _yes_. Having a UI way, like @adioo suggested for such things would be even better. Wordpress installation is a good example in this case.

**Comment by gabipetrovay** [comment]
There a are two things here:
1. How does the first application (i.e. `service`) gets on the server (I don't know yet)
2. How is service configured.

Indeed part 2 can be done by a self-configuration wizard like Wordpress does. But until we are there some scripts can help us move faster. The scripts I wrote are Bash/NodeJs hybrids so they can be improved to pure JS once as we move along.

For the part 1, there must be a chicken, or an egg somewhere. :smile: But we will see this as we go further. One can think also of a localhost `service` installation that deploys a service to a remote server (production). But still the egg, or the chicken moves on localhost. I cannot see it without an `"Enter"` being pressed.

**Comment by gabipetrovay** [comment]
@adioo After you `git pull; rm -Rf node_modules; npm install` in `service`, and if you get the error, see please what version of bash you are using. We can of course relax that rule (check if user is an email) if your machine does not support it.

Bash version you have: `bash --version`
What default shell you use: `cat /etc/passwd | grep "$USER" | cut -d : -f 7`

**Comment by IonicaBizau** [comment]
Shouldn't we move this conversation internally since it's service related?

**Comment by gabipetrovay** [comment]
How do you move a conversation? :stuck_out_tongue_winking_eye:

Anyway the whole topic is `service` related and has not much to do with login. Login only uses the infrastructure `service` provides.

**Comment by IonicaBizau** [comment]
> How do you move a conversation?

Close this issue and open a new one internally. :smile: 

> Anyway the whole topic is service related and has not much to do with login.

Exactly, so why to discuss such things here?

**Comment by ottiker** [comment]
@gabipetrovay strangly, now I have to provide a github username an password. And if I provide them it does nothing..

**Bash version:**

``` sh
GNU bash, version 4.3.30(1)-release (x86_64-pc-linux-gnu)
```

**Default shell:**

``` sh
/bin/bash
```

**Comment by IonicaBizau** [comment]
I don't get @adioo's error, however the email is not validated, even I provided a correct one.

**Comment by gabipetrovay** [comment]
Bash 4.0 vs 3.0 compatibility issue in regexp processing. Fixed in [50c301d](https://github.com/jillix/service/commit/50c301da9c7a6f89ce6f00fe79354bfe29a84f41).

Closing for now. See you on the other side. :wave:
